R/cite.r

```diff
@@ -16,6 +16,10 @@ cite <- function(x, ...) UseMethod("cite")
 #' cite(polymod)
 #' @export
 cite.survey <- function(x, ...) {
+  warning(
+    "The cite function is deprecated and will stop working in version 0.4.0. ",
+    "Please use get_citation() instead."
+  )
   chkDots(...)
 
```

Comment:
A simpler and more robust approach is to call `get_citation(x)` internally rather than duplicating the code